File: packages/maplibre/ios/Plugin/Classes/Camera.swift
import Capacitor
import Foundation
import MapLibre

public class Camera {
    let bearing: Double
    let center: LatLng
    let pitch: Double
    let zoom: Double

    init(_ mapView: MLNMapView) {
        self.bearing = mapView.direction
        self.center = LatLng(mapView.centerCoordinate)
        self.pitch = Double(mapView.camera.pitch)
        self.zoom = mapView.zoomLevel
    }

    func toJSObject() -> JSObject {
        var result = JSObject()
        result["bearing"] = bearing
        result["center"] = center.toJSObject()
        result["pitch"] = pitch
        result["zoom"] = zoom
        return result
    }
}

File: packages/maplibre/android/src/main/java/io/capawesome/capacitorjs/plugins/maplibre/classes/Camera.java
package io.capawesome.capacitorjs.plugins.maplibre.classes;

import androidx.annotation.NonNull;
import com.getcapacitor.JSObject;
import io.capawesome.capacitorjs.plugins.maplibre.MapLibreHelper;
import io.capawesome.capacitorjs.plugins.maplibre.interfaces.Result;
import org.maplibre.android.camera.CameraPosition;
import org.maplibre.android.geometry.LatLng;

public class Camera implements Result {

    private final double bearing;

    @NonNull
    private final LatLng center;

    private final double pitch;
    private final double zoom;

    public Camera(@NonNull CameraPosition position) {
        LatLng target = position.target;
        this.bearing = position.bearing;
        this.center = target == null ? new LatLng(0, 0) : target;
        this.pitch = position.tilt;
        this.zoom = position.zoom;
    }

    @Override
    @NonNull
    public JSObject toJSObject() {
        JSObject result = new JSObject();
        result.put("bearing", bearing);
        result.put("center", MapLibreHelper.createLatLngObject(center));
        result.put("pitch", pitch);
        result.put("zoom", zoom);
        return result;
    }
}
